src/components/sbb-datepicker/sbb-datepicker.tsx

```diff
@@ -0,0 +1,213 @@
+  }
+
+  /** Set the input value to the correctly formatted value. */
+  @Method() public async setValueAsDate(date: Date): Promise<void> {
+    const newValue: string = this._formatValue(
+      `${date.getDate()}.${date.getMonth() + 1}.${date.getFullYear()}`
+    );
+    await this._formatAndUpdateValue(newValue);
+  }
+
+  private _datePickerController: AbortController;
+
+  private _inputObserver = new MutationObserver(this._onInputPropertiesChange.bind(this));
+
+  private _dateAdapter = new NativeDateAdapter();
+
+  public connectedCallback(): void {
+    this._inputElement = getInput(this._element, this.input);
+  }
+
+  public componentDidLoad(): void {
+    this.inputUpdated.emit({
+      disabled: this._inputElement?.disabled,
+      readonly: this._inputElement?.readOnly,
+      min: this._inputElement?.min,
+      max: this._inputElement?.max,
+    });
+  }
+
+  public disconnectedCallback(): void {
+    this._inputObserver?.disconnect();
+    this._datePickerController?.abort();
+  }
+
+  private _formatValue(value: string): string {
+    const match: RegExpMatchArray = value?.match(REGEX);
+    if (!match || match.length <= 2) {
+      return value;
+    }
+
+    const day: string = match[1].padStart(2, '0');
+    const month: string = match[2].padStart(2, '0');
+    let year: number = +match[3];
+    if (year < 100 && year >= 0) {
+      year += 1900;
+    }
+    return `${day}.${month}.${year}`;
+  }
+
+  private _valueChanged(event): void {
+    this._formatAndUpdateValue(event.target.value);
+  }
+
+  /** Applies the correct format to values and triggers event dispatch. */
+  private async _formatAndUpdateValue(value: string): Promise<void> {
+    if (this._inputElement) {
+      this._inputElement.classList.remove('sbb-invalid');
+      this._inputElement.value = this._formatValue(value);
+      const newValueAsDate: Date = await this.getValueAsDate();
+      if (
+        !isDateAvailable(
+          newValueAsDate,
+          this._element.dateFilter,
+          this._inputElement?.min,
+          this._inputElement?.max
+        )
+      ) {
+        this._inputElement.classList.add('sbb-invalid');
+      }
+      this._emitChange();
+    }
+  }
+
+  /** Emits the change event. */
+  private _emitChange(): void {
+    this.change.emit();
+    this.didChange.emit();
+  }
+
+  private _onInputPropertiesChange(): void {
+    this.inputUpdated.emit({
+      disabled: this._inputElement?.disabled,
+      readonly: this._inputElement?.readOnly,
+      min: this._inputElement?.min,
+      max: this._inputElement?.max,
+    });
+  }
+
+  private _preventCharInsert(event): void {
+    const match = event.target.value.match(REGEX_PATTERN);
+    event.target.value = match?.[0] ?? null;
+  }
+
+  public render(): JSX.Element {
+    if (this._inputElement) {
+      this._inputElement.placeholder = i18nDatePickerPlaceholder[this._currentLanguage];
```

Comment:
I'd only overwrite if there is no placeholder on input (did we discuss this? :D)